src/Confluent.Kafka/ErrorCode.cs

```diff
@@ -29,6 +29,26 @@ namespace Confluent.Kafka
     public enum ErrorCode
     {
         /// <summary>
+        ///     Value serialization error
+        /// </summary>
+        Local_ValueSerialization = -1004,
```

Comment:
@edenhill - what do you think about the idea of reserving a block of librdkafka error return codes for arbitrary client use? alternatively these particular values could be defined by librdkafka even though it never produces them (it's reasonable other clients might also need them). 

the first idea has the disadvantage that different clients may define different values for the same error, which could potentially cause confusion. For that reason I'd prefer it if librdkafka defined them individually.

Maybe these error codes could be prefixed with `Client_` rather than `Local_` here (or something else). Maybe librdkafka could distinguish this type of error code (one that it never produces) using triple underscore.